src/Data/String/CodePoints.erl

```diff
@@ -28,12 +30,21 @@ end.
 '_singleton'(_Fallback) -> fun (CP) ->
     unicode:characters_to_binary([CP], utf8)
 end.
+
 '_take'(_Fallback) ->
     fun (N) ->
-        fun (S) ->
-            unicode:characters_to_binary(lists:sublist(unicode:characters_to_list(S, utf8), N), utf8)
-        end
+        fun (S) -> take(N, S, <<>>) end
     end.
+
+take(N, _S, Cs) when N =< 0 -> Cs;
+take(_N, <<>>, _Cs) -> <<>>;  % trying to take too many yields empty string
```

Comment:
pretty sure `take 5 "hi" == "hi"`

Reply:
The docs do appear to be incorrect, they contradict the PS implementation(s) and test, I have submitted a PR